pkg/arvo/tests/lib/bip32.hoon

```diff
@@ -0,0 +1,179 @@
+::  tests for the bip32 lib
+::
+::  Test vectors from:
+::  https://en.bitcoin.it/wiki/BIP_0032_TestVectors
+::
+/+  *test, bip32
+=,  bip32
+::
+|%
++$  vector  [mk=byts dp=tape id=@ux sk=@ux pk=@ux cc=@ux]
+::
+++  test-vectors
+  ^-  tang
+  |^  ;:  weld
+        %+  category  "identity"
+        (zing (turn vectors check-id))
+        %+  category  "public-key"
+        (zing (turn vectors check-pk))
+        %+  category  "private-key"
+        (zing (turn vectors check-sk))
+        %+  category  "chaincode"
+        (zing (turn vectors check-cc))
+      ==
+  ::
+  ++  check-id
+    |=  vector
+    %+  expect-eq
+      !>  id
+      !>  ^-  @ux
+         =,  (from-seed mk)
+         =,  (derive-path dp)
+         identity
```

Comment:
It's actually pretty cool that this works, but, IMO, it's an abuse of the `=,` namespace-ing system. It'd be better to be explicit about the stateful threading, something like:

```
=/  bip  (from-seed mk)
=.  bip  (derive-path.bip dp)
identity.bip
```

Or inline:

```
identity:(derive-path:(from-seed mk) dp)
```

Reply:
even better, I think, would be to factor out the derivation from the subtests. But I wasn't sure how to accomplish that.